src/main/java/at/hannibal2/skyhanni/features/rift/area/mountaintop/SunGeckoHelper.kt

```diff
@@ -0,0 +1,291 @@
+            val slot = modifier.slot
+            val item = event.inventoryItems[slot] ?: continue
+            if (item.item != Item.getItemFromBlock(Blocks.stained_glass_pane)) continue
+            if (item.itemDamage != 5) continue
+            modifiers.add(modifier)
+        }
+
+    }
+
+    @HandleEvent
+    fun onActionBar(event: ActionBarUpdateEvent) {
+        if (!isEnabled()) return
+        sunGeckoActionBar.findMatcher(event.actionBar) {
+            val firstHalf = groupOrNull("firstHalf") ?: return
+            val secondHalf = groupOrNull("secondHalf") ?: return
+            combo = groupOrNull("combo")?.toIntOrNull() ?: return
+
+            val firstHalfFull = firstHalf.count { it == '⬛' }
+            val secondHalfFull = secondHalf.count { it == '⬛' }
+            var comboHitCount = firstHalfFull + secondHalfFull
+            var totalHits = 10
+
+            if (!group().contains("§c")) {
+                timeSinceLastHit = SimpleTimeMark.now()
+            }
+
+            if (modifiers.contains(MODIFIERS.CULMINATION)) {
+                totalHits -= 1
+            }
+            if (modifiers.contains(MODIFIERS.TIME_SLICED)) {
+                if (timeSliceDuration.inWholeSeconds > 150) {
+                    totalHits -= 1
+                }
+            }
+
+            if (comboHitCount == 9 && totalHits == 8) {
+
+                // this is a hypixel bug
+                // it goes from 8/8 to 9/8 to 2/8
+                // the combo does not go up at 9/8
+                // so i guess the overlay is wrong but whatever
+                comboHitCount = 1
+            }
+            actionBarFormatted = "§a$comboHitCount/$totalHits"
+        }
+    }
+
+    @HandleEvent
+    fun onScoreboardUpdate(event: ScoreboardUpdateEvent) {
+        if (!isEnabled()) return
+        for (line in event.full) {
+            if (line.startsWith(" Big damage in: §d")) {
+                modifiers.add(MODIFIERS.TIME_SLICED)
+                timeSliceDuration = TimeUtils.getDuration(line.replace(" Big damage in: §d", ""))
+            }
+        }
+    }
+
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        if (!isEnabled()) return
+        if (event.message == "§f                           §r§c§lACTIVE MODIFIERS!") {
+            scanningChat = true
+            modifiers.clear()
+        } else if (event.message == "§6§l▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬") {
+            scanningChat = false
+        }
+
+        if (scanningChat) {
+            for (modifier in MODIFIERS.entries) {
+                if (event.message.contains(modifier.name.allLettersFirstUppercase())) {
+                    modifiers.add(modifier)
+                }
+            }
+        }
+    }
+
+    @HandleEvent
+    fun onAreaChanged(event: ScoreboardAreaChangeEvent) {
+        if (!isEnabled()) return
+        reset()
+        if (event.area == "Time Chamber") {
```

Comment:
wouldnt it be better to do this?
```suggestion
        inTimeChamber = event.area == "Time Chamber"
        if (!inTimeChamber) reset()
```